addons/compute_worker/gpu_uniforms/gpu_uniform.gd

```diff
@@ -138,22 +138,67 @@ func byte_array_to_vec3(array: PackedByteArray) -> Vector3:
 	return vec
 
 
-## Convert a Vector3 to GLSL equivalent `dvec3, dvec4` format
-func vec3_to_byte_array(vector: Vector3) -> PackedByteArray:
+## Convert a Vector3 to GLSL equivalent `dvec3` format
+func dvec3_to_byte_array(vector: Vector3) -> PackedByteArray:
 	
 	return PackedFloat64Array([vector.x, vector.y, vector.z, 0.0]).to_byte_array()
 
+## Convert a Vector3 to GLSL equivalent `vec3` format
+func vec3_to_byte_array(vector: Vector3) -> PackedByteArray:
+	
+	return PackedFloat32Array([vector.x, vector.y, vector.z, 0.0]).to_byte_array()
 
-## Convert a Vector3i to GLSL equivalent `ivec3, ivec4` format
+
+## Convert a Vector3i to GLSL equivalent `ivec3` format
 func vec3i_to_byte_array(vector: Vector3i) -> PackedByteArray:
 	
 	# We have to add a value for the "w" field for the vector,
 	# because the alignment spec for GLSL vec3s requires 16bytes
 	return PackedInt32Array([vector.x, vector.y, vector.z, 0]).to_byte_array()
 
+func vec2i_to_byte_array(vector: Vector2i) -> PackedByteArray:
+	
+	# We have to add a value for the "z" field for the vector,
+	# because the alignment spec for GLSL vec2s requires 16bytes
+	return PackedInt32Array([vector.x, vector.y, 0]).to_byte_array()
```

Comment:
The component in the `Z` position here is not necessary. `vec2`s in GLSL only need the `x` and `y` elements. As opposed to `vec3`s, which are treated as 4-component vectors by GLSL.